src/components/Form/EditProfileForm/EditProfileForm.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { Form, Formik } from "formik";
+import * as Yup from "yup";
+import { Button } from "design-react-kit";
+import { Link, useHistory } from "react-router-dom";
+import ProfileInfo from "../CreateProfileForm/ProfileData/ProfileInfo";
+import ReferentData from "../CreateProfileForm/ProfileData/ReferentData";
+import Api from "../../../api";
+import { DASHBOARD } from "../../../navigation/routes";
+import { RootState } from "../../../store/store";
+
+const EditProfileForm = () => {
+  const history = useHistory();
+  const agreementState = useSelector(
+    (state: RootState) => state.agreement.value
+  );
+  const [currentProfile, setCurrentProfile] = useState<any>();
+
+  useEffect(() => {
+    const getProfile = async (agreementId: string) => {
+      const response = await Api.Profile.getProfile(agreementId);
+      void setCurrentProfile(response.data);
+    };
+
+    void getProfile(agreementState.id);
+  }, []);
+
+  const editProfile = (profile: any) => {
+    if (agreementState) {
+      void Api.Profile.updateProfile(agreementState.id, profile);
+    }
+  };
```

Comment:
Can we introduce `TaskEither` in that cases?

Reply:
Yes, of course.